pydocstringformatter/_formatting/formatters_numpydoc.py

```diff
@@ -0,0 +1,135 @@
+
+
+class NumpydocNameColonTypeFormatter(NumpydocSectionFormatter):
+    """Ensure proper spacing around the colon separating names from types."""
+
+    name = "numpydoc-name-type-spacing"
+
+    numpydoc_sections_with_parameters = (
+        "Parameters",
+        "Attributes",
+        "Returns",
+        "Yields",
+        "Receives",
+        "Other Parameters",
+        "See Also",
+    )
+
+    def treat_sections(
+        self, sections: OrderedDict[str, list[str]]
+    ) -> OrderedDict[str, list[str]]:
+        """Ensure proper spacing around the colon separating names from types."""
+        new_sections = OrderedDict([])
+        for section_name, section_lines in sections.items():
+            if section_name in self.numpydoc_sections_with_parameters:
+                initial_indent = section_lines[0].index(section_name)
+                new_section = []
+                for line in section_lines:
+                    if len(line) <= initial_indent:
+                        # blank or just whitespace
+                        new_line = line
+                    elif line[initial_indent].isspace():
+                        # Description block
+                        new_line = line
+                    elif ":" not in line:
+                        # Header line, or no name, or maybe no
+                        # type/description
+                        new_line = line
+                    else:
+                        param_name, param_type = line.split(":", 1)
+                        new_line = f"{param_name.rstrip():s} : {param_type.lstrip():s}"
+                    new_section.append(new_line)
+            else:
+                new_section = section_lines
+            new_sections[section_name] = new_section
+        return new_sections
+
+
+class NumpydocSectionSpacingFormatter(NumpydocSectionFormatter):
+    """Ensure proper spacing between sections."""
+
+    name = "numpydoc-section-spacing"
+
+    def treat_sections(
+        self, sections: OrderedDict[str, list[str]]
+    ) -> OrderedDict[str, list[str]]:
+        """Ensure proper spacing between sections."""
+        new_sections = OrderedDict([])
+        for section_name, section_lines in sections.items():
+            last_line = section_lines[-1]
+            if not (last_line == "" or last_line.isspace()):
+                section_lines.append("")
+            new_sections[section_name] = section_lines
+        return new_sections
+
+
+class NumpydocSectionHeaderLineLengthFormatter(NumpydocSectionFormatter):
+    """Ensure section header lines are proper length."""
+
+    name = "numpydoc-section-header-line-length"
+
+    def treat_sections(
+        self, sections: OrderedDict[str, list[str]]
+    ) -> OrderedDict[str, list[str]]:
+        """Ensure section header lines are proper length."""
+        new_sections = OrderedDict([])
+        first_section = True
+        for section_name, section_lines in sections.items():
+            if first_section:
```

Comment:
But what if the first section isn't a summary? Could you add a test for this?

Reply:
If it starts with
```python
    """Parameters
    ----------
```
then there is no summary section; I think the current code would change this to
```python
    """
    Parameters
    ------------
```
To fix this further, I would need a consistent name for the summary/deprecation warning/extended summary section.

Reply:
See above.

Although I think this change is good, it should probably also be its own formatter. But let's add that in a follow up PR.

Reply:
I *think* I fixed this.  Should I change a test to make sure?

Reply:
Yes, I think that would be good!

Reply:
"Works" now, although interaction with the section ordering formatter can produce strange results:
```python
    """    Parameters
    ----------
    ...

Returns
    -------
    ...
```
I think there's a formatter to strip the leading space; not sure about indenting `Returns` properly.